NetKet/UnitaryTimeEvolution/TimeStepper/time_stepper.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef TIME_STEPPER_HPP
+#define TIME_STEPPER_HPP
+
+#include <memory>
+
+#include "Utils/json_helper.hpp"
+
+#include "explicit_time_steppers.hpp"
+#include "controlled_time_steppers.hpp"
+
+namespace netket { namespace ode {
+
+namespace detail {
+
+template<class Stepper>
+std::unique_ptr<Stepper> ControlledStepperFromJson(const json& pars, int dim)
+{
+    double atol = FieldOrDefaultVal(pars, "AbsTol", 1e-9);
+    double rtol = FieldOrDefaultVal(pars, "RelTol", 1e-9);
+    return std::unique_ptr<Stepper>(new Stepper(atol, rtol, dim));
+}
+
+template<class Stepper>
+std::unique_ptr<Stepper> ExplicitStepperFromJson(const json& pars, int dim)
+{
+    double dt = FieldVal(pars, "InternalTimeStep");
+    return std::unique_ptr<Stepper>(new Stepper(dt, dim));
+}
+
+}
+
+template<class State>
+std::unique_ptr<AbstractTimeStepper<State>>
```

Comment:
Returning `unique_ptr` from factory functions is safe and, e.g., recommended by Herb Sutter [in this blog post](https://herbsutter.com/2013/05/30/gotw-90-solution-factories/) which contains a nice discussion on why this is a good idea.

As far as I understand, what happens is the following: `std::unique_ptr` cannot be copied (if the pointer would be copied on return, this would indeed be unsafe as the original pointer would deallocate the object). However, returning it by value from this function actually results in the pointer being moved out of the function (which does not destroy its content) because of [copy elision](https://stackoverflow.com/questions/4316727/returning-unique-ptr-from-functions) and thus transferring ownership of the object to the calling code.